base.nix

```diff
@@ -35,4 +35,24 @@
     alsa.support32Bit = true;
     pulse.enable = true;
   };
+
+  # Backport mutter patch which fix flickering in scenarios where the dri
+  # driver is unavailable.
+  #
+  #   https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3117
+  #
+  nixpkgs.overlays = [
```

Comment:
Note: this will not currently work on the Snowfall Lib branch since pkgs is passed in to the nixosSystem call. This overlay will need to be extracted from the configuration here. We may also want to make it optional to avoid cache misses for folks who don't need it.

Ideally this patch gets included in NixPkgs itself.